celavi/routing.py

```diff
@@ -193,199 +191,125 @@ def get_all_routes(
-            # merge source and destination pairs
-            source_dest_pairs = _source_loc.merge(_dest_loc, on="merge")
-
-            # Filter down to only the source/destination pairs allowed by route_pairs
-            source_dest_allowable_pairs = route_pairs[
-                ["source_facility_type", "destination_facility_type"]
-            ].apply(tuple, axis=1)
-            source_dest_complete_pairs = source_dest_pairs[
-                ["source_facility_type", "destination_facility_type"]
-            ].apply(tuple, axis=1)
-            source_dest_select_route_types = source_dest_complete_pairs.isin(
-                source_dest_allowable_pairs
-            )
-            selected_routes = source_dest_pairs[source_dest_select_route_types]
-            all_route_list = selected_routes.merge(
-                route_pairs, on=["source_facility_type", "destination_facility_type"]
+        network = pd.read_csv(network_edges)
+
+        # Get a table of which processing steps require downstream
+        # landfill facilities, for use in postprocessing routes
+        _landfill_edges = network.loc[
+            [vkmtmax > 0 and instate 
+            for vkmtmax, instate in zip(network.vkmt_max, network.in_state)],
+            ['u_step','v_step']
+            ].drop_duplicates().reset_index()
+        
+        _network_dist = []
+
+        vkmt_by_county_list = []
+
+        if router is None:
+            print('No router found', flush = True)
+            
+        else:
+            print(f'Router.get_all_routes: Finding {len(network)} routes', flush = True)
+            _rtime = time.time()
+            for idx, edge in network.iterrows():                
+                # Do not find routes between co-located nodes
+                if edge.u_facility_id == edge.v_facility_id:
+                    _network_dist = _network_dist + [{'index': idx, 'vkmt': 0.0, 'route_id': 'colocated'}]
+                # Do not find routes between co-located facilities
+                elif edge.u_lat == edge.v_lat and edge.u_long == edge.v_long:
+                    _network_dist = _network_dist + [{'index': idx, 'vkmt': 0.0, 'route_id': 'colocated'}]
+                else:
+                    _vkmt_by_county = router.get_route(
+                        start=(
+                            edge.u_long,
+                            edge.u_lat,
+                        ),
+                        end=(
+                            edge.v_long,
+                            edge.v_lat,
+                        ),
+                    )
+                    # If the route distance is greater than a max distance, do not save
+                    # the route information. Instead, record the index to drop it from
+                    # the network routes file
+                    if _vkmt_by_county.vkmt.sum() > edge.vkmt_max:
+                        _network_dist = _network_dist + [{'index': idx, 'vkmt': _vkmt_by_county.vkmt.sum(), 'route_id': 'vkmt_max'}]
```

Comment:
I would like to understand that step more. I think it's robably to speed up and make the routes simpler but I would like to learn more about the logic.